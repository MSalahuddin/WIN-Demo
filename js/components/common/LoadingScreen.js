import React from 'react';
import Modal from 'react-native-modal';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import { SafeAreaView } from 'react-native';
import LottieView from 'lottie-react-native';

import { gamePlayLoading } from '../../../assets/gifs';
import {
  gamePlayAnimation
} from '../../../assets/animations';
const StyledModal = styled(Modal)`
  align-items: center;
  justify-content: flex-start;
  margin: 0;
`;
const GamePlayLoadingGif = styled.Image`
  height: 100%;
  width: 100%;
`;

const LoadingScreen = ({ isVisible }) => {
  if (isVisible) {
    return <LottieView source={gamePlayAnimation} autoPlay loop resizeMode="cover" />;
  }
  return null;
};

LoadingScreen.propTypes = {
  isVisible: PropTypes.bool.isRequired
};

export default LoadingScreen;