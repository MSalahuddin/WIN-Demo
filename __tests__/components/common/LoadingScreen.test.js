import React from 'react';
import { render } from 'react-native-testing-library';

import LoadingScreen from '../../../js/components/common/LoadingScreen';

describe('Component LoadingScreen', () => {
  test('renders correctly', () => {
    const output = render(<LoadingScreen isVisible />);
    expect(output).toMatchSnapshot();
  });
});
