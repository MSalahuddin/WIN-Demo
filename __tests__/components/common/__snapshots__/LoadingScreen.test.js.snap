// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LoadingScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
          ]
        }
      >
        <Image
          resizeMode="stretch"
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <Image
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {},
            ]
          }
        />
      </View>
    </View>
  </Modal>
</RCTSafeAreaView>
`;
